src/node_messaging.cc

```diff
@@ -153,7 +153,14 @@ class SerializerDelegate : public ValueSerializer::Delegate {
       : env_(env), context_(context), msg_(m) {}
 
   void ThrowDataCloneError(Local<String> message) override {
-    env_->isolate()->ThrowException(Exception::Error(message));
+    Local<Value> argv[] = {
+      message,
+      FIXED_ONE_BYTE_STRING(env_->isolate(), "DataCloneError")
+    };
+    env_->isolate()->ThrowException(
+        env_->domexception_function()
+            ->NewInstance(env_->context(), arraysize(argv), argv)
+            .ToLocalChecked());
```

Comment:
Using `.ToLocalChecked()` here is going to be problematic in the presence of `worker.terminate()`. It should be fine to just return if `NewInstance()` returns an empty handle

Reply:
Also, maybe add a `CHECK(!env_->domexception_function().IsEmpty());`?